### PR TITLE
Installer jre embedded

### PR DESCRIPTION
**Pull request by javydreamercsw**
Reques to to pull the installer changes to run from embedded JRE instead of the one installed in the target computer.

I believe I explained well enough my commits but feel free to contact me if any questions arise.

### CODE REVIEW

**Comment by javydreamercsw** [comment]
All commits appear to have been done in the last two days but I've been longer working on it. Just messed up my repository and had to recreate from local copies.

File: code/gradle/release.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@
  *
  * Release script
  * 1. gradle prepareRelease slowtest
- *   a.	Update version � remove snapshot
+ *   a.	Update version � remove snapshot
```
Comment (LegacyKing) [inline code/gradle/release.gradle]: Minor typo looks like you have an emdash or dash which displays as the weird question mark. The minus sign will display properly. This is repeated on line 20 as well.

File: code/gradle/release.gradle [hunk 2]
```diff
@@ -117,64 +118,144 @@ ext {
...
+    with baseData
+}
+
+//See: https://github.com/TheBoegl/gradle-launch4j\
+launch4j {
+    mainClassName = 'pcgen.system.Main'
+    icon = "${projectDir}/installers/win-installer/Local/pcgen.ico"
+    outfile = "pcgen.exe"
+    jreMinVersion = '1.8.0'
+    jreMaxVersion = '1.8.9'
```
Comment (LegacyKing) [inline code/gradle/release.gradle]: Should this be set to 1.8 before the code base has been moved to 1.8 compliance?

File: code/gradle/release.gradle [hunk 3]
```diff
@@ -248,27 +351,27 @@ task checksum(dependsOn: assembleArtifacts) << {
...
-	description="Release a new version of PCGen. This will build PCGen, produce the release artifacts, tag the release and then update and commit the version ready for new dev work."
+    description="Release a new version of PCGen. This will build PCGen, produce the release artifacts, tag the release and then update and commit the version ready for new dev work."
 }
 
 // Update the PCGen version to indicate a release, then commit the new version
 task updateVersionRelease << {
-	unSnapshotVersion()
-	commitFile(new File("gradle.properties"), "Update version for release ${project.version}")
+    unSnapshotVersion()
+    commitFile(new File("gradle.properties"), "Update version for release ${project.version}")
```
Comment (LegacyKing) [inline code/gradle/release.gradle]: I'm wondering if this actually does anything. When originally written, this was an SVN based design (tagging the release actually functioned). I'm not sure this is functioning now that we use git and github.com.

File: code/pcgen-launch4j.cfg.xml [hunk 4]
```diff
@@ -1,23 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
 <launch4jConfig>
   <dontWrapJar>true</dontWrapJar>
   <headerType>gui</headerType>
   <jar>pcgen.jar</jar>
-  <outfile>C:\Projects\pcgendev\code\pcgen.exe</outfile>
+  <outfile>Z:\Netbeans\pcgen\code\pcgen.exe</outfile>
```
Comment (LegacyKing) [inline code/pcgen-launch4j.cfg.xml]: This looks like a custom build change. Should this be using 'Netbeans' when the release monkeys use different IDEs?

File: code/pcgen-launch4j.cfg.xml [hunk 5]
```diff
@@ -1,23 +1,27 @@
...
+  <icon>Z:\Netbeans\pcgen\installers\win-installer\Local\pcgen.ico</icon>
   <jre>
-    <path></path>
-    <minVersion>1.6.0</minVersion>
-    <maxVersion>1.7.9</maxVersion>
+    <path>jre/</path>
+    <bundledJre64Bit>false</bundledJre64Bit>
+    <bundledJreAsFallback>false</bundledJreAsFallback>
+    <minVersion>1.8.0</minVersion>
+    <maxVersion>1.8.9</maxVersion>
```
Comment (LegacyKing) [inline code/pcgen-launch4j.cfg.xml]: Version set to 1.8 is premature (project still using 1.7 until @cpmeister updates code base to accept 1.8)

**Comment by LegacyKing** [comment]
Looks good to me other than the version setting and other minor nits. Just waiting on @thpr or @cpmeister to review and merge.

File: code/gradle/release.gradle (hunk 2 above)
Comment (javydreamercsw) [inline code/gradle/release.gradle]: I changed it due to the moving to JavaFX which needs Java 8+. That's where this change branched from. I can change it back and you guys update it when the rest is ready.

File: code/gradle/release.gradle (hunk 3 above)
Comment (javydreamercsw) [inline code/gradle/release.gradle]: Outside of my scope. Nothing for me to do here.

File: code/pcgen-launch4j.cfg.xml (hunk 4 above)
Comment (javydreamercsw) [inline code/pcgen-launch4j.cfg.xml]: This file changes when you use it with launch4j jar. This file is not used by the build script and will be removed once testing of the script generated one are complete.

I do use Netbeans and I'm aware others do as well as I see related questions/comments on the email traffic.

**Comment by javydreamercsw** [comment]
Reverted some unnecessary changes and adjusted as per your comments.

**Comment by LegacyKing** [comment]
@cpmeister or @thpr been two weeks...

**Comment by javydreamercsw** [comment]
I fixed those observations already, long time ago...

**Comment by cpmeister** [comment]
@javydreamercsw can you double check that all files use the unix line endings. It looks like build.gradle and a few of the other ones you are checking in are using the windows line endings.

**Comment by javydreamercsw** [comment]
I ran a command to do that on all the sources and many files showed up and were fixed. Only committing the ones I modified.

**Comment by javydreamercsw** [comment]
I could commit the rest, but this seems way outside of my current scope. As FYI, I used dos2unix tool as per this question: http://stackoverflow.com/questions/7068179/convert-line-endlings-for-whole-directory-tree-git

**Comment by cpmeister** [comment]
You don't need to go that far, I was only concerned about the files in this PR. Sorry for taking so long to resolve this. Thanks for the great work.

**Pull request merged**